light-client-verifier/src/verifier.rs

```diff
@@ -111,119 +111,160 @@ where
             hasher,
         }
     }
-}
-
-impl<P, C, V, H> Verifier for PredicateVerifier<P, C, V, H>
-where
-    P: VerificationPredicates,
-    C: VotingPowerCalculator,
-    V: CommitValidator,
-    H: Hasher,
-{
-    /// Validate the given light block state.
-    ///
-    /// - Ensure the latest trusted header hasn't expired
-    /// - Ensure the header validator hashes match the given validators
-    /// - Ensure the header next validator hashes match the given next validators
-    /// - Additional implementation specific validation via `commit_validator`
-    /// - Check that the untrusted block is more recent than the trusted state
-    /// - If the untrusted block is the very next block after the trusted block, check that their
-    ///   (next) validator sets hashes match.
-    /// - Otherwise, ensure that the untrusted block has a greater height than the trusted block.
-    ///
-    /// **NOTE**: If the untrusted state's `next_validators` field is `None`,
-    /// this will not (and will not be able to) check whether the untrusted
-    /// state's `next_validators_hash` field is valid.
-    fn verify(
-        &self,
-        untrusted: UntrustedBlockState<'_>,
-        trusted: TrustedBlockState<'_>,
-        options: &Options,
-        now: Time,
-    ) -> Verdict {
-        // Ensure the latest trusted header hasn't expired
-        verdict!(self.predicates.is_within_trust_period(
-            trusted.header_time,
-            options.trusting_period,
-            now,
-        ));
-
-        // Ensure the header isn't from a future time
-        verdict!(self.predicates.is_header_from_past(
-            untrusted.signed_header.header.time,
-            options.clock_drift,
-            now,
-        ));
 
+    /// Validates an `UntrustedBlockState`.
+    pub fn validate(&self, untrusted: &UntrustedBlockState<'_>) -> Result<(), VerificationError> {
         // Ensure the header validator hashes match the given validators
-        verdict!(self.predicates.validator_sets_match(
+        self.predicates.validator_sets_match(
             untrusted.validators,
             untrusted.signed_header.header.validators_hash,
             &self.hasher,
-        ));
+        )?;
 
         // TODO(thane): Is this check necessary for IBC?
         if let Some(untrusted_next_validators) = untrusted.next_validators {
             // Ensure the header next validator hashes match the given next validators
```

Comment:
Yes, the check is necessary so we can remove the comment
```suggestion
        // Ensure the header next validator hashes match the given next validators
        if let Some(untrusted_next_validators) = untrusted.next_validators {
```